src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs

```diff
@@ -2112,6 +2112,48 @@ public void PresentHeaders_ChainHeaderTreeAdvances_UnconsumedBlocksDataBytes_Equ
             Assert.Equal(0, chainedHeaderTree.UnconsumedBlocksDataBytes);
         }
 
+        /// <summary>
+        /// Issue 45 @ Node has a reorg of 5 blocks.
+        /// Make sure that headers for the blocks that were reorged away
+        /// have no block pointers and block data availability == headers only.
+        /// </summary>
+        [Fact]
+        public void NodeHasReorg_ReorganisedHeaders_HaveNoBlockPointers_BlockDataAvailability_SetToHeadersOnly()
+        {
+            const int depthOfReorg = 5;
+            const int heightOfFork = 5;
+            const int initialChainSize = 10;
+            const int extensionSize = 20;
+
+            TestContext testContext = new TestContextBuilder().WithInitialChain(initialChainSize).Build();
+            ChainedHeaderTree chainedHeaderTree = testContext.ChainedHeaderTree;
+            ChainedHeader chainTip = testContext.InitialChainTip;
+
+            ChainedHeader forkedBlockHeader = chainTip.GetAncestor(depthOfReorg);
+            ChainedHeader newTip = testContext.ExtendAChain(extensionSize, forkedBlockHeader);
+            List<BlockHeader> listOfExtendedHeaders = testContext.ChainedHeaderToList(newTip, heightOfFork + extensionSize);
+            ConnectNewHeadersResult connectNewHeadersResult = chainedHeaderTree.ConnectNewHeaders(1, listOfExtendedHeaders);
+
+            foreach (ChainedHeader chainedHeader in connectNewHeadersResult.ToHashArray())
+            {
+                chainedHeaderTree.BlockDataDownloaded(chainedHeader, newTip.FindAncestorOrSelf(chainedHeader).Block);
+                chainedHeaderTree.PartialValidationSucceeded(chainedHeader, out bool fullValidationRequired);
+                chainedHeaderTree.ConsensusTipChanged(chainedHeader);
+            }
+
+            // Check status of headers for the blocks that were reorged away.
+            ChainedHeader chainHeader = chainTip;
+            while (chainHeader.Height > forkedBlockHeader.Height)
+            {
+                // Header has no block pointer.
+                Assert.Null(chainHeader.Block);
+
+                // Header block data availability == headers only.
+                Assert.Equal(BlockDataAvailabilityState.HeaderOnly, chainHeader.BlockDataAvailability);
+                chainHeader = chainHeader.Previous;
+            }
+        }
```

Comment:
Looks quite good, I would just like to have a second peer to claim the original chain so that it is not removed from the tree when we fork away.So all I'm asking for is to have a second peer to present headers for the original chain before we present the fork.

Reply:
@Aprogiena Made changes - have I understood them correctly?